orx-palette/src/main/kotlin/PaletteStudio.kt

```
@@ -0,0 +1,219 @@
+) : Extension {
+    var palettes: MutableList<MutableList<ColorRGBa>> = mutableListOf()
+    var palette: Palette = defaultPalette
+
+    private var paletteIndex: Int = 0
+
+    enum class SortBy {
+        NO_SORTING, DARKEST, BRIGHTEST
+    }
+
+    enum class Collections {
+        ONE, TWO, THREE
+    }
+
+    private val collectionsResource = mapOf(
+        Collections.ONE to getCollPath("1"),
+        Collections.TWO to getCollPath("2"),
+        Collections.THREE to getCollPath("3")
+    )
+
+    var background: ColorRGBa? = null
+        get() {
+            return palette.background
+        }
+    var foreground: ColorRGBa? = null
+        get() {
+            return palette.foreground
+        }
+    var colors: Colors? = null
+        get() {
+            return palette.colors
+        }
+    var colors2: Colors? = null
+        get() {
+            return palette.colors2
+        }
+
+    init {
+        if (loadDefault) {
+            loadCollection(collection)
+        }
+    }
+
+    private fun loadCollection(newCollection: Collections) {
+        val collectionPath: URL = collectionsResource[newCollection]!!
+
+        palettes = mutableListOf()
+
+        loadFromURL(collectionPath)
+
+        val choice = Random.pick(palettes)
+
+        palette = createPalette(choice)
+        paletteIndex = palettes.indexOf(choice)
+    }
+
+    private fun load(contents: String) {
+        try {
+            val gson = GsonBuilder().create()
+
+            val clipData = gson.fromJson(
+                    contents, Array<Array<String>>::class.java
+            ).toList()
+
+            for (p in clipData) {
+                val palette = mutableListOf<ColorRGBa>()
+
+                for (colorStr in p.toList()) {
+                    val colorInt = fromHex(colorStr)
+
+                    palette.add(colorInt)
+                }
+
+                palettes.add(palette)
+            }
+        } catch (ex: Exception) {
```

It will not stop the program from running, I remember @edwinRNDR saying something about failing fast. Maybe it's better to rethrow the exception?

---

I would keep the error messages, because they help the user figuring out where and what could the issue be. Regarding re-throwing the exception I'm not against it. Let @edwinRNDR add is 2 cents.